### PR TITLE
Enable confirmation for producers (#660)

### PR DESCRIPTION
**Pull request by amigian74**
Use confirmation acknowledgment for producer

### CODE REVIEW

File: RabbitMq/Producer.php [hunk 1]
```diff
@@ -41,15 +46,37 @@ protected function getBasicProperties()
...
      *
      * @param string $msgBody
      * @param string $routingKey
      * @param array $additionalProperties
-     * @param array $headers
+     * @param array|null $headers
+     * @return bool
      */
-    public function publish($msgBody, $routingKey = null, $additionalProperties = array(), array $headers = null)
+    public function publish($msgBody, $routingKey = '', $additionalProperties = array(), array $headers = null)
```
Comment (Eloar) [inline RabbitMq/Producer.php:79]: why would you change routing key from `null` to `''`?

**Review by Eloar — CHANGES_REQUESTED** [review]
Your test fails, you should review it

**Comment by ruudk** [comment]
can you please give this a proper title?

File: RabbitMq/Producer.php [hunk 2]
```diff
@@ -40,15 +45,37 @@ protected function getBasicProperties()
...
      *
      * @param string $msgBody
      * @param string $routingKey
      * @param array $additionalProperties
-     * @param array $headers
+     * @param array|null $headers
+     * @return bool
      */
-    public function publish($msgBody, $routingKey = null, $additionalProperties = array(), array $headers = null)
+    public function publish($msgBody, $routingKey = '', $additionalProperties = array(), array $headers = null): bool
```
[the remark this answers is not in the document]
Comment (amigian74) [reply RabbitMq/Producer.php]: These changes are not from me - I only changed the return type for the function publish

File: RabbitMq/Producer.php (hunk 1 above)
Comment (amigian74) [reply RabbitMq/Producer.php:79]: These changes are not from me - I only changed the return type for the function publish

**Comment by amigian74** [comment]
Happy new year. What is missing here?

**Comment by Eloar** [comment]
@amigian74 as far as I can tell there was requested change to ProducerInterface (to not to declare return type with type hint as it is breaking change). I was suggesting adding description to doc comment of added method.

**Comment by amigian74** [comment]
Hi. In my opinion all requests should by fullfilled by now

**Comment by amigian74** [comment]
How ist this possible? I only added a doc comment to the new function.

**Comment by amigian74** [comment]
Can anybody please help me with this - this feature is really important for us

**Comment by amigian74** [comment]
Is there still something to do here?

**Comment by amigian74** [comment]
Hi, are you planning a new release soon with this request in it? It's really important for me,

**Pull request closed without merging**